sources/packages/backend/apps/api/src/services/institution/institution.service.ts

```diff
@@ -683,14 +683,25 @@ export class InstitutionService extends RecordDataModelService<Institution> {
   /**
    * Get the institution by ID.
    * @param institutionId Institution id.
+   * @param options related options.
+   * `entityManager` entity manager to be used in the transaction.
    * @returns Location retrieved, if found, otherwise returns null.
    */
-  async getInstitutionDetailById(institutionId: number): Promise<Institution> {
-    return this.repo
+  async getInstitutionDetailById(
+    institutionId: number,
+    options?: { entityManager?: EntityManager },
+  ): Promise<Institution> {
+    const repo =
```

Comment:
This can be simplified as below since we are using the base class in the old way for this class.
```ts
const repo =  options?.entityManager?.getRepository(Institution) ?? this.repo;
```